src/Elasticsearch.Net/Connection/MetaData/MetaDataHeaders.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+namespace Elasticsearch.Net
+{
+	internal sealed class MetaDataHeaders : IMetaDataHeaders
```

Comment:
both `internal` and `sealed` are used very sparingly. Siding on the side of being public by default has some benefits.

Not sure we need `IMetaDataHeaders` if this class is public? It also allows us to unit  test this in isolation.

Reply:
In this case, I think I probably agree 😄. I tend to lean the other way, internal and sealed until they need to be since we can change them freely without breaking changes to consumers. But when I've done that in other libs, it can cause some headaches for testing. Here, I think loosing the interface is better.

Reply:
@Mpdreamz In the latest update to this PR I've been able to avoid exposing this on the configuration entirely. Therefore I've left this internal now since it's used only within the provider.